src/Impostor.Server/Net/State/Game.Incoming.cs

```diff
@@ -156,11 +156,13 @@ private async ValueTask<GameJoinResult> AddClientSafeAsync(ClientBase client)
             if (_compatibilityConfig.AllowVersionMixing == false &&
                 this.Host != null && client.GameVersion != this.Host.Client.GameVersion)
             {
-                if (client.GameVersion < this.Host.Client.GameVersion)
+                _compatibilityManager.TryGetCompatibilityGroup(client.GameVersion, out var clientCompatGroup);
```

Comment:
I know all clients should have a compat group at this point but it still feels wrong to just ignore the return value

Reply:
Yeah, technically you could race this using the API, so I should check here again if I'm not comparing against null